Dockerfile.builder

```diff
@@ -45,9 +45,22 @@ RUN apt-get update \
 		zip \
 		zlib1g-dev \
 	&& apt-get clean \
-	&& rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/* \
-	&& update-alternatives --install /usr/bin/clang clang /usr/bin/clang-7 100 \
-	&& update-alternatives --install /usr/bin/llc llc /usr/bin/llc-7 100
+	&& rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
+
+#
+# Install clang/llvm
+#
+RUN git clone --depth 1 -b master https://github.com/llvm/llvm-project.git llvm && \
+	mkdir -p llvm/llvm/build/install && \
+	cd llvm/ && \
+	git checkout -b d941df363d1cb621a3836b909c37d79f2a3e27e2 d941df363d1cb621a3836b909c37d79f2a3e27e2 && \
+	cd llvm/build && \
+	cmake .. -G "Ninja" -DLLVM_TARGETS_TO_BUILD="BPF" -DLLVM_ENABLE_PROJECTS="clang" -DBUILD_SHARED_LIBS=OFF -DCMAKE_BUILD_TYPE=Release -DLLVM_BUILD_RUNTIME=OFF && \
+	ninja && \
+	cp bin/clang /usr/bin/clang-git && \
+	cp bin/llc /usr/bin/llc-git && \
+	update-alternatives --install /usr/bin/clang clang /usr/bin/clang-git 100 && \
+	update-alternatives --install /usr/bin/llc llc /usr/bin/llc-git 100
```

Comment:
I think it would be better to move this section into a dedicated docker image. The clang build seems to take too many resources which might cause the build time to increase unnecessarily. We could build clang separately and have a dedicated image for it.

Reply:
Also, we need to update the developer documentation that states that we now depend on this particular clang version.

Reply:
> We could build clang separately and have a dedicated image for it as we won't change clang version frequently but we change the image builder frequently.

Makes sense, was also thinking as another option to leave the builder at clang-7 since it doesn't need to load anything, but otoh, having both versions aligned might be better. I can check once done with the rest. Docs also once the other stuff is resolved in here.